doc/manual/src/glossary.md

```diff
@@ -85,12 +85,16 @@
 
     [store path]: #gloss-store-path
 
+  - [file system object]{#gloss-store-object}\
+    The Nix data model for representing file system data.
+
+    See [File System Object](@docroot@/architecture/file-system-object.md) for details.
+
+    [file system object]: #gloss-file-system-object
+
   - [store object]{#gloss-store-object}\
-    A file that is an immediate child of the Nix store directory. These
-    can be regular files, but also entire directory trees. Store objects
-    can be sources (objects copied from outside of the store),
-    derivation outputs (objects produced by running a build task), or
-    derivations (files describing a build task).
+
+    A store object consists of a [file system object], [reference]s to other store objects, and other metadata.
```

Comment:
We forgot the obvious.

```suggestion
    A store object consists of a [file system object], [reference]s to other store objects, and other metadata.
    It can be referred to by [store path].
```

Reply:
I think it needs to be

> they can be referred to by store path

or

> it can be referred to be a store path